src/uk/gov/hmcts/contino/GithubAPI.groovy

```diff
@@ -13,6 +13,11 @@ class GithubAPI {
     this.steps = steps
   }
 
+  private cachedLabelList = [
```

Comment:
shouldn't this be static so it persists across the whole pipeline run? in case someone is creating a new instance at a different point?

Reply:
My understanding is that a static property is decoupled from the instance.  Therefore if we make it static, there will be one cache for all instances of the githubapi class within that pipeline, which we don't want (because each instance could point to a different repo).  Multiple instances of the API should not be created for the same repo within a given pipeline - the original api object should be reused instead.

Reply:
yes but there's no way for someone to get access to it outside this pipeline step we use so they would have to create their own instance in their Jenkinsfile. not sure why you would check another repo other than your own

Reply:
Ah I see - being unable to pass the original API object does make the case for static more compelling.  I'm not 100% sure if it will work as intended.  We should test with and without.

(WRE other repos, some PR's will stand up multiple pods for different applications.  I was thinking that if those additional pods are based on PR's it might be useful/necessary to check the labels on those also - probably a fringe use case though.)

Reply:
we do the static approach elsewhere, from my understanding static is for a pipeline run not Jenkins itself so it should be fine, but worth testing with 2 different repositories